functions/pipeline.py

```diff
@@ -0,0 +1,467 @@
+
+    # Get sample id that maps between metadata and expression files
+    map_experiment_sample = metadata[[sample_id_colname]]
+
+    # Get all unique experiment ids
+    experiment_ids = np.unique(np.array(map_experiment_sample.index)).tolist()
+    print("There are {} experiments in the compendium".format(
+        len(experiment_ids)))
+
+    # Get only sample ids that have expression data available
+    sample_ids_with_gene_expression = list(normalized_data.index)
+
+    # Get associated experiment ids with expression data
+    experiment_ids_with_gene_expression = []
+
+    for experiment_id in experiment_ids:
+
+        if "Human" in dataset_name:
+            # Some project id values are descriptions
+            # We will skip these
+            if len(experiment_id) == 9:
+                selected_metadata = metadata.loc[experiment_id]
+                sample_ids = list(selected_metadata[sample_id_colname])
+
+                if any(x in sample_ids_with_gene_expression for x in sample_ids):
+                    experiment_ids_with_gene_expression.append(experiment_id)
+        else:
+            selected_metadata = metadata.loc[experiment_id]
+            sample_ids = list(selected_metadata[sample_id_colname])
+
+            if any(x in sample_ids_with_gene_expression for x in sample_ids):
+                experiment_ids_with_gene_expression.append(experiment_id)
+
+    print('There are {} experiments with gene expression data'.format(
+        len(experiment_ids_with_gene_expression)))
+
+    # Save file with experiment ids
+    experiment_ids_with_gene_expression_df = pd.DataFrame(
+        experiment_ids_with_gene_expression, columns=['experiment_id'])
+    experiment_ids_with_gene_expression_df.to_csv(output_file, sep='\t')
+    print('{} experiment ids saved to file'.format(
+        len(experiment_ids_with_gene_expression)))
+
+    del map_experiment_sample, selected_metadata
+    gc.collect()
+
+
+def train_vae(config_file,
+              input_data_file):
+    '''
+    Trains VAE model using parameters set in config file
+
+    Arguments
+    ----------
+    config_file: str
+        File containing user defined parameters
+
+    input_data_file: str
+        File path corresponding to input dataset to use
+
+    '''
+
+    # Read in config variables
+    params = utils.read_config(config_file)
+
+    # Load parameters
+    base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+    dataset_name = params['dataset_name']
+    learning_rate = params['learning_rate']
+    batch_size = params['batch_size']
+    epochs = params['epochs']
+    kappa = params['kappa']
+    intermediate_dim = params['intermediate_dim']
+    latent_dim = params['latent_dim']
+    epsilon_std = params['epsilon_std']
+    train_architecture = params['NN_architecture']
+
+    # Read data
+    normalized_data = pd.read_table(
+        input_data_file,
+        header=0,
+        sep='\t',
+        index_col=0).T
+
+    print(normalized_data.shape)
```

Comment:
```suggestion
```

Reply:
Updated the print statement to act as a check for the user